clients/src/main/java/org/apache/kafka/common/requests/FindCoordinatorRequest.java

```diff
@@ -110,19 +111,18 @@ public static FindCoordinatorRequest parse(ByteBuffer buffer, short version) {
 
     @Override
     protected Struct toStruct() {
-        short version = version();
-        Struct struct = new Struct(ApiKeys.FIND_COORDINATOR.requestSchema(version));
-        if (version == 0) {
+        Struct struct = new Struct(ApiKeys.FIND_COORDINATOR.requestSchema(version()));
+        if (struct.hasField(GROUP_ID_KEY_NAME))
             struct.set(GROUP_ID_KEY_NAME, coordinatorKey);
-        } else {
-            struct.set(COORDINATOR_TYPE_KEY_NAME, coordinatorType.id);
+        else
             struct.set(COORDINATOR_KEY_KEY_NAME, coordinatorKey);
-        }
+        if (struct.hasField(COORDINATOR_TYPE_KEY_NAME))
+            struct.set(COORDINATOR_TYPE_KEY_NAME, coordinatorType.id);
         return struct;
     }
 
     public enum CoordinatorType {
-        GROUP((byte) 0), TRANSACTION((byte) 1), UNKNOWN((byte) -1);
+        GROUP((byte) 0), TRANSACTION((byte) 1);
```

Comment:
Hmm. So I actually use the `UNKNOWN` value in my transactional producer patch to indicate that a particular request doesn't need to be sent to a coordinator. But I guess that's not a strong enough reason to keep this.